lib/core/connector_settings.rb

```diff
@@ -32,6 +32,11 @@ def self.fetch(connector_id)
       new(es_response, globals)
     end
 
+    def initialize(es_response, globals)
+      @elasticsearch_response = es_response.with_indifferent_access
+      @globals = globals.with_indifferent_access
```

Comment:
Also about `with_indifferent_access` - I've encountered this problem only when in tests, do you remember how it happened for you?

Reply:
I just encountered this problem in dev:

```
swiftype(dev)> Core::ElasticConnectorActions.connectors({ term: { is_native: true } }).first.index_name
Traceback (most recent call last):
       16: from railties (5.2.8.1) lib/rails/commands.rb:18:in `<main>'
       15: from railties (5.2.8.1) lib/rails/command.rb:46:in `invoke'
       14: from railties (5.2.8.1) lib/rails/command/base.rb:69:in `perform'
       13: from thor (0.20.3) lib/thor.rb:387:in `dispatch'
       12: from thor (0.20.3) lib/thor/invocation.rb:126:in `invoke_command'
       11: from thor (0.20.3) lib/thor/command.rb:27:in `run'
       10: from railties (5.2.8.1) lib/rails/commands/console/console_command.rb:96:in `perform'
        9: from railties (5.2.8.1) lib/rails/commands/console/console_command.rb:19:in `start'
        8: from railties (5.2.8.1) lib/rails/commands/console/console_command.rb:64:in `start'
        7: from org/jruby/RubyKernel.java:1237:in `catch'
        6: from org/jruby/RubyKernel.java:1237:in `catch'
        5: from org/jruby/RubyKernel.java:1507:in `loop'
        4: from org/jruby/RubyKernel.java:1091:in `eval'
        3: from (irb):1:in `evaluate'
        2: from connectors_utility (8.5.0.0.pre.20220912T090241Z) lib/core/connector_settings.rb:45:in `index_name'
        1: from connectors_utility (8.5.0.0.pre.20220912T090241Z) lib/core/connector_settings.rb:41:in `[]'
NoMethodError (undefined method `[]' for nil:NilClass)
```

Reply:
> @vidok , `Core::ElasticConnectorActions.connectors`, is this a new method you defined locally?
> 
> Hi @artem-shelkovnikov , I believe it's here:
> 
> https://github.com/elastic/connectors-ruby/blob/233288027638c3a5212b3d0d4d98428290db91c3/lib/core/elastic_connector_actions.rb#L71
> 
> you will get `nil` for `hit[:_source]` but `hit['_source']` works fine.

It's not local anymore https://github.com/elastic/connectors-ruby/blob/main/lib/core/elastic_connector_actions.rb#L53. 

I made it to support different query terms when looking up connector documents.